plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/xsd/com.google.cloud.tools.eclipse.appengine.libraries.xsd

```diff
@@ -0,0 +1,267 @@
+will determine how to resolve the entries associated with the container.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+         <xs:attribute name="name" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  A user-friendly name for the library that can be displayed in the UI, e.g.: "AppEngine Endpoints".
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+         <xs:attribute name="siteUri" type="xs:string">
+            <xs:annotation>
+               <xs:documentation>
+                  URI that can be used to display a link to the user to provide more detailed information about the library, e.g. user guides, wiki, other documentation.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="libraryFile">
+      <xs:annotation>
+         <xs:documentation>
+            Describes a single artifact located in a Maven repository.
+         </xs:documentation>
+      </xs:annotation>
+      <xs:complexType>
+         <xs:sequence>
+            <xs:element ref="mavenCoordinates"/>
+            <xs:sequence minOccurs="0" maxOccurs="unbounded">
+               <xs:choice minOccurs="0" maxOccurs="1">
+                  <xs:element ref="exclusionFilter" />
+                  <xs:element ref="inclusionFilter" />
+               </xs:choice>
+            </xs:sequence>
+         </xs:sequence>
+         <xs:attribute name="export" type="xs:boolean" use="required">
+            <xs:annotation>
+               <xs:documentation>
+                  If export attribute is true, then the corresponding file will be included in the WAR file of the
+project, under WEB-INF/lib.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+
+         <xs:attribute name="javadocUri" type="xs:anyURI">
+            <xs:annotation>
+               <xs:documentation>
+                  URI that can be used by the IDE to lookup javadoc for the classes contained in this artifact.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+         <xs:attribute name="sourceUri" type="xs:anyURI">
+            <xs:annotation>
+               <xs:documentation>
+                  URI that can be used by the IDE to lookup sources for the classes contained in this artifact.
+               </xs:documentation>
+            </xs:annotation>
+         </xs:attribute>
+      </xs:complexType>
+   </xs:element>
+
+   <xs:element name="exclusionFilter">
+      <xs:annotation>
+         <xs:documentation>
+            A pattern that can be used to exclude classes and packages to be visible. Uses the same format as the JDT's build path inclusion/exclusion filters: http://ant.apache.org/manual/dirtasks.html#patterns
```

Comment:
This should link to [`IClasspathEntry#getExclusionPatterns()`](http://help.eclipse.org/mars/topic/org.eclipse.jdt.doc.isv/reference/api/org/eclipse/jdt/core/IClasspathEntry.html#getExclusionPatterns--) (or `IAccessRule`, I suppose) as`IClasspathEntry#getExclusionPatterns()` notes:

> The pattern mechanism is _similar_ to Ant's.

(emphasis mine)